test_scripts/pypeit_tests.py

```diff
@@ -420,6 +423,24 @@ def run(self):
         else:
             return super().run()
 
+
+class PypeItVet(PypeItTest):
+    """Test subclass that runs vet "unit" tests"""
+
+    def __init__(self, setup, pargs):
+        super().__init__(None, None, None)
+        self.passed = True
+
+    def build_command_line(self):
+        return ''
+
+    def run(self):
+        # TODO -- figure out how to run unit tests from here
+        pass
+        #tmp = pytest.main(["-x", os.path.join(self.env['PYPEIT_DEV'], 'vet_tests')])
```

Comment:
"tmp" in this case can be treated as an int. aka `if tmp != 0` will tell if the test failed. Trickier is getting stdout to a logfile for later viewing. I recommend creating  a log with get_unique_file, and then doing:
```
In [12]: with open(logfile, "w") as f:
    ...:     with contextlib.redirect_stdout(f):
    ...:         s = pytest.main(['-x', self.tests])
    ...:     self.passed == (s == 0)
```